test/evergreen/print_stack_trace.py

```diff
@@ -109,49 +109,58 @@ def dump(self, exe_path: str, core_path: str, lib_path: str, dump_all: bool, out
 def main():
     parser = argparse.ArgumentParser()
     parser.add_argument('-e', '--executable_path',
-                        help='path to the executable',
-                        required=True)
+                        help='path to the executable')
     parser.add_argument('-c', '--core_path',
-                        help='directory path to the core dumps',
-                        required=True)
+                        help='directory path to the core dumps')
     parser.add_argument('-l', '--lib_path', help='library path')
-    group = parser.add_mutually_exclusive_group(required=True)
-    group.add_argument('--unit_test', action='store_true', help='Format core dumps from python unit tests')
-    group.add_argument('--format', action='store_true', help='Format core dumps from format tests')
     args = parser.parse_args()
 
     # Store the path of the core files as a list.
     core_files = []
-    dump_all = args.unit_test
+    regex = re.compile(r'dump.*core', re.IGNORECASE)
+    core_path = args.core_path
+    if core_path is None:
+        core_path = '.'
 
-    regex = None
-    if (args.format):
-        regex = re.compile(r'dump_t.*core', re.IGNORECASE)
-    elif (args.unit_test):
-        regex = re.compile(r'dump.*python.*core', re.IGNORECASE)
-
-    for root, _, files in os.walk(args.core_path):
+    for root, _, files in os.walk(core_path):
         for file in files:
             if regex.match(file):
                 core_files.append(os.path.join(root, file))
 
     for core_file_path in core_files:
         print(border_msg(core_file_path), flush=True)
+
+        # If not specified, derive the binary name from the core dump folder.
+        # The test/format and cpp suites are different and need specific handling. All the other
+        # tests have an executable that follows the pattern "test_<test_folder>". For example,
+        # the test folder wt2853_perf contains an executable named test_wt2853_perf.
+        if args.executable_path is None:
+            # Retrieve the first part of the path without the core name.
+            dirname = core_file_path.rsplit('/', 1)[0]
+            if 'cppsuite' in core_file_path:
+                executable_path = dirname + "/run"
+            elif 'format' in core_file_path:
+                executable_path = dirname + "/t"
+            else:
+                test_name = dirname.rsplit('/', 1)[1]
+                executable_path = dirname + "/test_" + test_name
+        else:
+            executable_path = args.executable_path
```

Comment:
assuming that we catch the python3 cases here?

Reply:
Let me double-check that, I am afraid my testing was incorrect. The code does not seem to handle Python tests correctly.

Reply:
It was not working, thanks for pointing it out.
Unfortunately, we use different Python binaries in our testing suite, so it is not possible to "guess" it from the script, thus we have to make sure the Python binary is specified when we find a core dump from a Python test.
I am specifying the Python binary in the commit a1e70b58d4a877e8d9d5ca953e6642da0acc40b0 and the processing is done in this commit 961136c8ba39af0944c9f1ad86d4e94772f4f7e3.